File: litho-lint-detector/src/main/java/com/github/pavlospt/utils/PsiUtils.java
package com.github.pavlospt.utils;

import com.intellij.psi.PsiAnnotation;
import com.intellij.psi.PsiMethod;
import com.intellij.psi.PsiModifierList;
import com.intellij.psi.PsiParameterList;

public class PsiUtils {

  public static boolean hasAnnotations(PsiModifierList psiModifierList) {
    if (psiModifierList == null) return false;

    PsiAnnotation[] psiAnnotations = psiModifierList.getAnnotations();

    return psiAnnotations.length > 0;
  }

  public static boolean hasParameters(PsiMethod psiMethod) {
    if (psiMethod == null) return false;

    PsiParameterList psiParameterList = psiMethod.getParameterList();

    return psiParameterList.getParametersCount() > 0;
  }
}

File: litho-lint-detector/src/main/java/com/github/pavlospt/utils/PsiUtils.kt
package com.github.pavlospt.utils

import com.intellij.psi.PsiMethod
import com.intellij.psi.PsiModifierList
import com.intellij.psi.PsiParameterList

object PsiUtils {

  @JvmStatic
  fun hasAnnotations(
      psiModifierList: PsiModifierList?): Boolean = psiModifierList?.annotations?.isNotEmpty() ?: false

  @JvmStatic
  fun hasParameters(
      psiMethod: PsiMethod?): Boolean {
    val psiParameterList: PsiParameterList? = psiMethod?.parameterList
    val parametersCount = psiParameterList?.parametersCount ?: 0
    return parametersCount > 0
  }
}
